### PR TITLE
Update dependencies, tooling and workflows

### PR DESCRIPTION
**Pull request by hews**
Add GitHub actions and get testing and linting running correctly.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+
+%.lint:
+	cd $* && go vet ./... && golangci-lint run ;
 
-test: ## Run tests.
-	go test -short -v ./... -race;
-	cd middleware && go test -short -v ./... -race;
-	cd logrus && go test -short -v ./... -race;
-	cd zerolog && go test -short -v ./... -race;
+%.test:
```
Comment (iaburton) [inline Makefile:16]: We should probably have a 'flake' or 'test-flake' target that does similar to this but adds the failfast option along with a test count of 100 or so.

Comment (hews) [reply Makefile:16]: I'll add that to the CI/CD: do we also need to run it during local dev?

Comment (iaburton) [reply Makefile:16]: Na, I think CICD is fine.

**Review by iaburton — APPROVED** [review]
# :partying_face:

**Pull request merged**